src/Tribe/Main.php

```diff
@@ -1175,6 +1176,94 @@ public function do_addons_api_settings_tab() {
 			include_once $this->plugin_path . 'src/admin-views/tribe-options-addons-api.php';
 		}
 
+		/**
+		 * should we show the upgrade nags?
+		 */
+		public function show_upgrade() {
+			$show_tab = current_user_can( 'activate_plugins' );
+
+			/**
+			 * Provides an oppotunity to override the decision to show or hide the upgrade tab
+			 *
+			 * Normally it will only show if the current user has the "activate_plugins" capability
+			 * and there are some currently-activated premium plugins.
+			 *
+			 * @var bool
+			 */
+			if ( ! apply_filters( 'tribe_events_show_upgrade_tab', $show_tab ) ) {
+				return false;
+			}
+
+			/**
+			 * This constant was introduced in the View Alpha
+			 * for agencies or others who'd like to hide the V2 Views upgrade prompt.
+			 * let them set a constant in wp-config that will be respected
+			 */
+			if ( defined( 'TRIBE_HIDE_V2_VIEWS_UPGRADE' ) ) {
```

Comment:
Can we add a function on this file similar to the one we have there and replace the direct use of the constant?
https://github.com/moderntribe/the-events-calendar/blob/master/src/functions/views/provider.php

Reply:
this _is_ a function - it checks the constant and a filter.  Is there value to putting it elsewhere?

Reply:
I don't find value in putting it elsewhere until a new need arises.